### DISCUSSION TITLE
How to check results

### DISCUSSION THREAD

**Discussion by chri2** [rspamd/rspamd#4696] [General]
```[ZERO_WIDTH_SPACE_URL(7.00){bücher.de;},PHISHING(3.67){bücher.de->buecher.de;}```

I had an ham mail rejected because of the above ratings. I tried to search for _'ZERO_WIDTH_SPACE_URL'_ and just found one [issue](https://github.com/rspamd/rspamd/issues/4290) related to it. I suspect that the german umlauts did influence the rating here - maybe not as intended.

The second symbol seems to say that the german umlaut is shown somewhere instead of the correct domain name with the german 'ü' replaced by 'ue'. This might be a valid reason for the score if I interpret that correctly, but the intention of the sender has been good: make the mail a bit more readable.

I'd appreciate help to find more background informations and a way to test the `ZERO_WIDTH_SPACE_URL` symbol related to the german umlaut(s).

**Comment by fatalbanana** [comment]
You'll want hands on the actual message (or URL) in question- umlaut doesn't seem to be the culprit.

[Phishing](https://rspamd.com/doc/modules/phishing.html) module is looking for a hyperlink that masquerades as a different domain, like this:
~~~
<a href="http://evil.co.uk">http://example.co.uk</a>
~~~

**Reply by chri2** [reply]
> You'll want hands on the actual message

That's what I thought, but because of the high rating from ZERO_WIDTH_SPACE_URL (which I still do not understand - is there no documentation for this one?) the message got rejected. I whitelisted the domain it came from and hope that the next one will come through.

**Reply by fatalbanana** [reply]
It should mean that the URL contained a [zero-width space](https://en.wikipedia.org/wiki/Zero-width_space).

https://github.com/rspamd/rspamd/blob/2ac2b26de6105a64c89812671dffa687d8efe790/rules/misc.lua#L141
https://github.com/rspamd/rspamd/blob/2ac2b26de6105a64c89812671dffa687d8efe790/src/libserver/url.c#L1349
